lib/puppet/provider/package/windows/exe_package.rb

```diff
@@ -55,7 +57,26 @@ def match?(resource)
     end
 
     def self.install_command(resource)
-      munge(resource[:source])
+      if Puppet::FileSystem.exist?(resource[:source])
+        munge(resource[:source])
+      else
+        begin
+          Puppet.debug ("The source %{resource_source} does not exist locally. Using it as URI ") % { resource_source: resource[:source]}
+          uri = URI.parse(resource[:source])
```

Comment:
We have to watch out for special characters in `source`. If `source => 'http://www.example.com/a b`, then we need to escape the source before parsing like:

```ruby
URI(Puppet::Util.uri_encode(resource[:source]))
```

Otherwise, we could say the source needs to be URL escaped in the manifest:

```
source => 'http://www.example/a%20b'
```

I think the first option is more user-friendly.

Reply:
We should also reject valid URI's if the scheme is not `http` or `https`